code-challenges.test.js

```diff
@@ -15,42 +15,161 @@
+    // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
-const fibLength2 = 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    // Use 'describe ()', 'it ()' and 'expect ()' to test the syntax with jest.
+
+    describe ("function", () => {
+    const fibLength1 = 6
+    // Expected output: [1, 1, 2, 3, 5, 8]
+    const fibLength2 = 10
+    // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+
+    it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
+    expect(fib(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+    expect(fib(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+})
+})
+
+        // after running yarn jest received
+        // --> ReferenceError: fib is not defined
 
 
 // b) Create the function that makes the test pass.
 
+    // The Fibonacci sequence is a set of integers (the Fibonacci numbers) that starts with a zero, followed by a one, then by another one, and then by a series of steadily increasing numbers. The sequence follows the rule that each number is equal to the sum of the preceding two numbers.
+    // Create a function that if an 
 
-// --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
-// a) Create a test with expect statements for each of the variables provided.
+    const fib = function(n) {
+    if (n === 2) {
+    return [1, 1];
+    } else {
+    const fibLength1 = fib(n - 1);
+    fibLength1.push(fibLength1[fibLength1.length - 1] + fibLength1[fibLength1.length - 2]);
+    return fibLength1;
+    }
+};
 
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+        // **It took a long time but i finally figured it out. At the very bottom is the passing test. 
+    // function did not pass. The actual output did not equal the expected output. The output includes 7 elements instead of 6 and 11 elements instead of 10. Both arrays include a 0 and I could not figure out how to remove it. Tried .splice(), .pop() but no luck.
+    // console.log(fib(fibLength1)); output --> [ 0, 1, 1, 2, 3, 5, 8 ]
+            // Error received after running yarn jest
+    // expect(received).toEqual(expected) // deep equality
+    // - Expected  - 0
+    // + Received  + 1
+    // @@ -1,6 +1,7 @@
+    // Array [
+    // +   0,
 
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+    // after beating myself up for a VERY long time i found the problem. It's supposed to be a number greater than 2 however, i had 1. Also my return was [0,1] instead of [1,1]. Now output is correct --> [1,1,2,3,5,8]
 
+    // after running yarn jest received
+    // PASS  ./code-challenges.test.js
+    // function
+    // ✓ takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence (3 ms)
```

Comment:
I am impressed You got this code to work! You have nested functions that are recursive. Thats some high level stuff. 

I noticed you don't have pseudo code for this problem though. Your pseudo code starts on line 44 but never completes. Please remember pseudo code is required See line 4. I am send this back to you to complete your pseudo code.